go/stellar/stellarsvc/frontend.go

```diff
@@ -781,8 +781,72 @@ func (s *Server) GetSendAssetChoicesLocal(ctx context.Context, arg stellar1.GetS
 		return res, err
 	}
 
-	// Not implemented. CORE-8087
-	return res, fmt.Errorf("GetSendAssetChoicesLocal not implemented")
+	owns, err := stellar.OwnAccount(ctx, s.G(), arg.From)
+	if err != nil {
+		return res, err
+	}
+	if !owns {
+		return res, fmt.Errorf("account %s is not owned by current user", arg.From)
+	}
+
+	ourBalances, err := s.remoter.Balances(ctx, arg.From)
+	if err != nil {
+		return res, err
+	}
+
+	res = []stellar1.SendAssetChoiceLocal{}
+	for _, bal := range ourBalances {
+		asset := bal.Asset
+		if asset.IsNativeXLM() {
+			// We are only doing non-native assets here.
+			continue
+		}
+		choice := stellar1.SendAssetChoiceLocal{
+			Asset:   asset,
+			Enabled: true,
+			Left:    bal.Asset.Code,
+			Right:   bal.Asset.Issuer,
+		}
+		res = append(res, choice)
+	}
+
+	if arg.To != "" {
+		uis := libkb.UIs{
+			IdentifyUI: s.uiSource.IdentifyUI(s.G(), arg.SessionID),
+		}
+		mctx := s.mctx(ctx).WithUIs(uis)
+
+		recipient, err := stellar.LookupRecipient(mctx, stellarcommon.RecipientInput(arg.To))
+		if err != nil {
+			return res, err
```

Comment:
I think that this should not be fatal. It's ok to accidentally enable an asset since the send will be rejected later on by another layer. Log and return.